waspc/src/Wasp/Project/Analyze.hs

```diff
@@ -69,8 +84,57 @@ analyzeWaspProject waspDir options = do
   where
     fileNotFoundMessage = "Couldn't find the *.wasp file in the " ++ toFilePath waspDir ++ " directory"
 
-analyzeWaspFile :: Psl.Schema.Schema -> Path' Abs File' -> IO (Either [CompileError] [AS.Decl])
-analyzeWaspFile prismaSchemaAst waspFilePath = do
+analyzeWaspFile :: Psl.Schema.Schema -> WaspFile -> IO (Either [CompileError] [AS.Decl])
+analyzeWaspFile prismaSchemaAst = \case
+  WaspLangFile waspFilePath -> analyzeWaspLangFile prismaSchemaAst waspFilePath
+  WaspTsFile waspFilePath -> analyzeWaspTsFile prismaSchemaAst waspFilePath
+
+analyzeWaspTsFile :: Psl.Schema.Schema -> Path' Abs File' -> IO (Either [CompileError] [AS.Decl])
+analyzeWaspTsFile _prismaSchemaAst waspFilePath = do
+  let workingDir = SP.parent waspFilePath
+  chan <- newChan
+  (_, tscExitCode) <-
+    concurrently
+      (readJobMessagesAndPrintThemPrefixed chan)
+      (runNodeCommandAsJob workingDir "npx" ["tsc", "-p", "tsconfig.node.json"] J.Wasp chan)
+
+  case tscExitCode of
+    ExitFailure _status -> return $ Left ["Error while running TypeScript compiler on the *.wasp.mts file."]
+    ExitSuccess -> do
+      otherChan <- newChan
+      (_, runExitCode) <-
+        concurrently
+          (handleRunJsMessages otherChan)
+          ( runNodeCommandAsJob
+              workingDir
+              "node"
+              [ SP.fromAbsDir workingDir ++ "node_modules/wasp-ts-sdk/dist/run.js",
+                SP.fromAbsDir workingDir ++ ".wasp/config/main.wasp.mjs"
```

Comment:
This won't work on Win, because workingDir will have Win separators and you have posix separators on the right.
In general I recommend converting from StrongPath at the last moment possible, that will save you from mistakes like this.
So in this case:
```hs
SP.fromAbsFile $ workingDir </> [relfile|node_modules/wasp-ts-sdk/dist/run.js|],
...
```